src/commands/synthetics/reporters/junit.ts

```diff
@@ -481,22 +514,23 @@ export class JUnitReporter implements Reporter {
       properties: {
         property: [
           {$: {name: 'check_id', value: test.public_id}},
-          ...(isDeviceIdSet(result.result)
+          ...(hasResult(result) && isDeviceIdSet(result.result)
             ? [
                 {$: {name: 'device', value: result.result.device.id}},
                 {$: {name: 'width', value: result.result.device.width}},
                 {$: {name: 'height', value: result.result.device.height}},
               ]
             : []),
           {$: {name: 'execution_rule', value: test.options.ci?.executionRule}},
-          {$: {name: 'location', value: result.location}},
+          {$: {name: 'location', value: hasResult(result) && result.location}},
           {$: {name: 'message', value: test.message}},
           {$: {name: 'monitor_id', value: test.monitor_id}},
           {$: {name: 'passed', value: `${passed}`}},
           {$: {name: 'public_id', value: test.public_id}},
           {$: {name: 'result_id', value: result.resultId}},
           {$: {name: 'result_url', value: resultUrl}},
-          ...('startUrl' in result.result ? [{$: {name: 'start_url', value: result.result.startUrl}}] : []),
+          {$: {name: 'selective_rerun', value: renderSelectiveRerun(result.selectiveRerun)}},
+          {$: {name: 'start_url', value: hasResult(result) && 'startUrl' in result.result && result.result.startUrl}},
```

Comment:
ℹ️  There is a filter below, which filters out any falsy `$.value`:

https://github.com/DataDog/datadog-ci/blob/1d8e7821d132076a3237dc86f3bb185a355f3814/src/commands/synthetics/reporters/junit.ts#L532-L538